### DISCUSSION TITLE
How does debounce work in output apps?

### DISCUSSION THREAD

**Discussion by 101100** [datalust/seq-tickets#1784] [Help] [answered]
I am trying to understand how debounce works and it is not documented and the last discussion with exactly this title has been deleted for some reason (moved platforms?).

**Comment by nblumhardt** [comment, marked as answer]
Hi Jason! The debounce window works thus:

 * When an event reaches the app instance (and matches its input signal)
   - The event is processed by the app
   - The debounce window begins
 * During the debounce window, no further events are processed by the app instance
 * When the debounce window elapses, the process begins afresh

I think to answer your most likely question - events reaching the app _don't_ extend the debounce window if one is already active.

Hope this helps,
Nick

**Reply by 101100** [reply]
OK, so the events in the middle are just discarded, correct? Not bundled or collected in any way?

**Reply by nblumhardt** [reply]
That's correct :-) 👍

**Comment by Sebbl22** [comment]
Hi,
I would appreciate it, if a description for the Debounce window would be added to Seq. It seems, that for all other fields a description is present already.
![grafik](https://github.com/user-attachments/assets/6fdcb6f4-d78c-4125-acbf-84aef19a2722)


I was installing the "First of Type" app and I mistook the Debounce Windows for an app-specific setting - instead of a default Seq setting.

**Reply by nblumhardt** [reply]
Thanks for pointing this out! There was some help text in the page source, but a markup problem was causing it to be stripped during the build process. Fixed in the next 2025.1 build:

![image](https://github.com/user-attachments/assets/d9f4b1fa-d623-4cf3-be40-536fd9a0102e)
